### ISSUE TITLE
Where to put array in model.predict() function

### ISSUE THREAD

**Issue by kallols** [tensorflow/tfjs#178]
I have imported a pre trained keras model in my jvascript. I want to put array into the model to get he output. 
To convert the array to sensor I am taking the help from **model.predict(tf.ones([8, 10]), {batchSize: 4}).print();** written in tfjs tutorial.

But where and how should I  keep my array inside this model.predict() function.

I have to insert array([[0., 0., 0., ..., 0., 0., 0.]]) with (1,30000) shape.  I tried following line but couldn't get the result: 
`console.log(model.predict(tf.ones([1, 30000]), Array(b)));`
Please tell me how to do it so I can get the output.

**Comment by dsmilkov** [comment]
Hi,

The second parameter of [model.predict](https://js.tensorflow.org/api/0.8.0/#tf.Model.predict) is a config with the following keys:
- batchSize (number) Batch size (Integer). If unspecified, it will default to 32. Optional
- verbose (boolean) Verbosity mode. Defaults to false. Optional

See the [API ref](https://js.tensorflow.org/api/0.8.0/#tf.Model.predict) for more info.

**Comment by kallols** [comment]
@dsmilkov But I need to put the input array somewhere. where to put the input array. The keras model is already trained with batchsize and epochs. Why do I need to assign them again.

I just need to put the input array to the existing keras model I imported through json and get the output.

**Comment by caisq** [comment]
@kallols I assume you are asking how to send input data to `Model.predict()`. If my understanding is correct, the answer is the input data is just the first input argument to `Model.predict()`.

The example you cited: `model.predict(tf.ones([8, 10]), {batchSize: 4}).print()`, is just sending a dummy input made of all ones. In real applications, like yours, you'll replace `tf.ones([8, 10])`, with something else that has the correct shape.

Let me know if you have any remaining questions.

**Comment by kallols** [comment]
@caisq I have data in [1,30000] shape lets call it A. Do I need to convert this javascript array into tensorflow array using tf.convert_to_tensor() first ? or I can simply use model.predict(A, {batchSize: 4}) to get the output ?

**Comment by xam-ps** [comment]
In which browser did you try out your code? I experienced a problem with importing a keras model in Chrome (on Ubuntu 16.04) - the predict function didn't work. In Firefox everything works like a charm.

**Comment by kallols** [comment]
@xam-ps I am using chrome in windows. 
here I am sending the input to a pre trained keras model imported via json file. the input of the keras model in python is np.array([[1,1,1,.......,1,1,1]])
The code for javascript is below:

var newoutput=model.predict((tf.ones([1, 30000])));
var g=newoutput[0];
var h=newoutput[1];
console.log(g);
console.log(h);
console.log(newoutput);
I am getting output as undefined for g and h.

For newoutput I am getting following result.

e {isDisposed: false, size: 1, shape: Array(2), dtype: "float32", strides: Array(1), …}

The result shoulf be an array of shape (1,1) and will have valid number not undefined.

Where am I doing the mistake? could you please guide me.

**Comment by xam-ps** [comment]
@kallols To the the data from the tensor I'm using the [data()](https://js.tensorflow.org/api/0.8.0/#tf.Tensor.data) function. For your case this should be sth. like:

`
newoutput.data().then(function(label){
    console.log(label[0]);
    console.log(label[1]);
});
`
